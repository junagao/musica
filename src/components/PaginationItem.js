import React from 'react';
import PropTypes from 'prop-types';

import './PaginationItem.scss';

const PaginationItem = ({ itemName, btState, pageNumber, onChangePage }) => (
  <li className={`page-item ${btState}`}>
    <a
      onClick={() => onChangePage(pageNumber)}
      href="!#"
      className={`page-link ${btState}`}
    >
      {itemName}
    </a>
  </li>
);

PaginationItem.propTypes = {
  pageNumber: PropTypes.number,
  btState: PropTypes.string,
  onChangePage: PropTypes.func,
  itemName: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
};

PaginationItem.defaultProps = {
  btState: '',
  onChangePage: () => {},
  pageNumber: 0,
};

export default PaginationItem;
