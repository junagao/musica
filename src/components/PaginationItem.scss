@import '../variables';

.page-link {
  margin-right: 0.25rem;
  border-radius: 0.25rem;
  border: 1px solid $color-lighter;
  outline: none;
  cursor: pointer;
  min-width: 2rem;
  height: 2rem;
  line-height: 2rem;
  text-align: center;
  transition: all ease-out 0.3s;
  font-size: 1rem;
  display: block;
  &:hover {
    color: #fff;
    border: 1px solid $color-main;
    background-color: $color-main;
    transition: all ease-in 0.3s;
  }
}

.disabled a {
  border-color: #d9d9d9;
  color: #d9d9d9;
  cursor: not-allowed;
  pointer-events: none;
}

.active a {
  color: #fff;
  border: 1px solid $color-main;
  background-color: $color-main;
  transition: all ease-in 0.3s;
}