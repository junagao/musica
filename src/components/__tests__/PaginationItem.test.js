import React from 'react';
import { shallow } from 'enzyme';

import PaginationItem from 'components/PaginationItem';

describe('PaginationItem', () => {
  let wrapper;
  const itemNameMock = '1';
  const btStateMock = 'active';
  const pageNumberMock = 1;
  const onChangePageMock = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <PaginationItem
        itemName={itemNameMock}
        btState={btStateMock}
        pageNumber={pageNumberMock}
        onChangePage={onChangePageMock}
      />,
    );
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one `a` element with corresponded href prop and className `page-link`', () => {
    expect(wrapper.find('a').length).toEqual(1);
    expect(wrapper.find('a').prop('href')).toEqual('!#');
    expect(wrapper.find('a').hasClass('page-link')).toBeTruthy();
  });

  it('should render the pages numbers', () => {
    expect(wrapper.find('a').text()).toEqual('1');
  });

  it('should call onChangePage prop passing the page number', () => {
    const number = 1;
    wrapper.find('a').simulate('click', number);
    expect(onChangePageMock).toBeCalledWith(1);
  });
});
