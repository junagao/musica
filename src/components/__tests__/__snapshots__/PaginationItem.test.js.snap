// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationItem should render correctly 1`] = `
<li
  className="page-item active"
>
  <a
    className="page-link active"
    href="!#"
    onClick={[Function]}
  >
    1
  </a>
</li>
`;
